### ISSUE TITLE
Large inline Markdown visualization causes OOM

### ISSUE THREAD

**Issue by jackwhelpton** [kubeflow/pipelines#5513]
### Environment

*  KFP version: 1.0.1

### Steps to reproduce

Output a markdown artifact containing a large amount of Markdown inline:

```python
        metadata = {
          "outputs": [
          {
            "storage": "inline",
            "source": "<large markdown string: ~20MB>",
            "type": "markdown",
          }]
        }
```

### Expected result

The Visualizations tab and Run output should render the Markdown.

Instead, the original markup code is visible: instead, after some time loading, the Chrome tab will crash with SBOX_FATAL_MEMORY_EXCEEDED.

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

**Comment by jackwhelpton** [comment]
I'm going to experiment with uploading this Markdown content to GCS, and see if that is handled better.

**Comment by Bobgy** [comment]
Hi @jackwhelpton, this seems like an expected limitation on the UI side.

1. I'm curious why a markdown file needs to be 20MB+, can people actually read it in KFP UI?
2. we might be able to optimize UI code to specifically optimize for this case, but there's a limit we can achieve, I wonder what would be your requirement of the largest file you want to visualize.

**Comment by jackwhelpton** [comment]
I suspected that would be the case: I've worked around it by including only a limited amount of the data in the markdown that is displayed, pushing the full report out to GCS and including a link after the abridged version.

To give more background, what I'm displaying is a list of warnings/errors detected with a data set, kind of a "lint" for training and eval data. I was going to try committing the markdown and seeing if GitHub can display it, just as a test for what the browser itself can handle.

It might suffice to have something in the documentation that explains roughly where these limits lie: I'm not sure it's vital to fix this. The combination of "here are the first 50 errors" and "here's where to get a complete log file" is probably the way to go here.
